dashboard/app/models/level_group.rb

```diff
@@ -87,4 +88,135 @@ def pages
   def plc_evaluation?
     levels.map(&:class).uniq == [EvaluationMulti]
   end
+
+  # Surveys: How many students must complete a survey before any results are shown.
+  SURVEY_REQUIRED_SUBMISSION_COUNT = 5
+
+  # Surveys: Given a sublevel, and the known response to it, return a result object.
+  def self.get_sublevel_result(sublevel, sublevel_response)
+    sublevel_result = {}
+    if sublevel_response
+      case sublevel
+      when TextMatch, FreeResponse
+        sublevel_result[:result] = sublevel_response["result"]
+        sublevel_result[:type] = "free_response"
+      when Multi
+        student_result = sublevel_response["result"].split(",").sort.join(",")
+        # unless unsubmitted
+        unless student_result == "-1"
+          answer_text = sublevel.properties.try(:[], "answers").try(:[], student_result.to_i).try(:[], "text")
+          sublevel_result[:result_text] = answer_text
+          # Convert "0,1,3" to "A, B, D" for teacher-friendly viewing
+          sublevel_result[:result] = student_result.split(',').map{ |k| Multi.value_to_letter(k.to_i) }.join(', ')
+          sublevel_result[:type] = "multi"
+        end
+      end
+    end
+    sublevel_result
+  end
+
+  # Surveys: Returns all survey results given a student and a script level of type LevelGroup.
+  def self.get_survey_results_for_student(student, script_level)
+    student_results = {}
+
+    # Skip student if they haven't submitted for this LevelGroup.
+    user_level = student.user_level_for(script_level, script_level.level)
+    return unless user_level.try(:submitted)
+
+    last_attempt = student.last_attempt(script_level.level)
+    response = last_attempt.try(:level_source).try(:data)
+    return unless response
+
+    response_parsed = JSON.parse(response)
+
+    # Go through each sublevel
+    script_level.level.levels.each_with_index do |sublevel, sublevel_index|
```

Comment:
Cleaner as `each_with_index.map`?  (Then can return the mapped values and remove `student_results` local.)